homeassistant/components/onvif/util.py

```diff
@@ -6,6 +6,43 @@
 
 from zeep.exceptions import Fault
 
+from .models import Event
+
+
+def build_event_entity_names(events: list[Event]) -> dict[str, str]:
+    """Build entity names for events, with index appended for duplicates.
+
+    When multiple events share the same base name, a sequential index
+    is appended to distinguish them (sorted by UID).
+
+    Args:
+        events: List of events to build entity names for.
+
+    Returns:
+        Dictionary mapping event UIDs to their entity names.
+
+    """
+    # Group events by name
+    events_by_name: dict[str, list[Event]] = {}
```

Comment:
You could simplify this with `collections.defaultdict(list)` so you can always call append in the loop